### PR TITLE
Feat/stepper

### PR DESCRIPTION
**Pull request by vlacher12**

### CODE REVIEW

File: src/components/form/stepper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    const lessThanAllowed = currentValue < minValue;
+
+    React.useEffect(() => {
+      if (onChange && !moreThenAllowed && !lessThanAllowed) {
+        onChange(currentValue);
+      }
+    }, [currentValue]);
+
+    const increment = () => {
+      setValue(Number(currentValue) + step);
```
Comment (klevytskyi) [inline src/components/form/stepper/index.tsx:57]: currentValue should be already `number` why to cast it again? 🤔

File: src/components/form/stepper/index.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    };
+
+    const decrement = () => {
+      setValue(Number(currentValue) - step);
+    };
+
+    const handleChange = (event) => {
+      setValue(event.target.value);
```
Comment (klevytskyi) [inline src/components/form/stepper/index.tsx:65]: That's the right place to cast, as well catch NaN and also filter out non number inputs

File: src/components/form/stepper/style.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
+import { styled } from 'linaria/react';
+import { ThemeMode } from 'utils/types';
+
+export const StyledStepperInput = styled.div`
```
Comment (klevytskyi) [inline src/components/form/stepper/style.ts:4]:
```suggestion
export const StyledStepperInput = styled.div<{
mode: ThemeMode,
theme: Theme
}>`
```

you should use generics for this types of the components

**Review by klevytskyi — CHANGES_REQUESTED** [review]

**Comment by klevytskyi** [comment]
переделывайте 🔱

**Comment by snanovskyi** [comment]
giphy

**Comment by dimaponomarchuk** [comment]
Salam

**Comment by mainstreamer** [comment]
image

**Pull request merged**